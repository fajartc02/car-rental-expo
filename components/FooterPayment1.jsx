import { Ionicons } from "@expo/vector-icons";
import { TouchableOpacity, StyleSheet } from "react-native";
import { Col, Row } from "./Grid";
import { ThemedView } from "./ThemedView";
import { ThemedText } from "./ThemedText";
import formatCurrency from "@/utils/formatCurrency";
import React from "react";

export default function FooterPayment1({
  data,
  paymentIdSelected,
  onChangeStep,
}) {
  return (
    <ThemedView style={styles.footer}>
      <Row>
        <Col
          style={{
            flexDirection: "row",
            textAlign: "left",
            gap: 8,
          }}
        >
          <ThemedText style={{ fontWeight: "bold" }}>
            {formatCurrency(data?.price)}
          </ThemedText>
          <Ionicons name="chevron-down" size={24} color="black" />
        </Col>
      </Row>
      <Row>
        <Col style={{ width: "95%" }}>
          {!paymentIdSelected ? (
            <TouchableOpacity
              style={{
                ...styles.btnPayment,
                backgroundColor: "#C9E7CA",
              }}
              disabled
            >
              <ThemedText style={styles.textBtn}>Bayar</ThemedText>
            </TouchableOpacity>
          ) : (
            <TouchableOpacity
              style={styles.btnPayment}
              onPress={() => onChangeStep()}
            >
              <ThemedText style={styles.textBtn}>Bayar</ThemedText>
            </TouchableOpacity>
          )}
        </Col>
      </Row>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  footer: {
    alignItems: "flex-start",
    padding: 16,
    justifyContent: "flex-start",
    width: "100%",
    backgroundColor: "#EEEEEE",
    gap: 8,
  },
  textBtn: {
    color: "white",
    fontWeight: "bold",
  },
  btnPayment: {
    backgroundColor: "#3D7B3F",
    borderRadius: 8,
    padding: 12,
    alignItems: "center",
    justifyContent: "center",
  },
});
