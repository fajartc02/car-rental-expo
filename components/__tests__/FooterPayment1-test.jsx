import * as React from "react";
import renderer from "react-test-renderer";
import FooterPayment1 from "../FooterPayment1";

it(`FooterPa renders correctly`, () => {
  const submitOrder = () => {
    console.log("submitOrder");
  };
  const tree = renderer
    .create(
      <FooterPayment1
        data={{
          price: 1000000,
        }}
        paymentIdSelected={1}
        onChangeStep={() => jest.fn()}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
