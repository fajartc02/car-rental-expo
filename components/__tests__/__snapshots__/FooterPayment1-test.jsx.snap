// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterPa renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "flex-start",
        "backgroundColor": "#EEEEEE",
        "gap": 8,
        "justifyContent": "flex-start",
        "padding": 16,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "baseline",
        "flexDirection": "row",
        "gap": 0,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
          "textAlign": "left",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontWeight": "bold",
            },
          ]
        }
      >
        IDRÂ 1,000,000
      </Text>
      <Text />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "baseline",
        "flexDirection": "row",
        "gap": 0,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "width": "95%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#3D7B3F",
            "borderRadius": 8,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "white",
                "fontWeight": "bold",
              },
            ]
          }
        >
          Bayar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
